src/Network/Affjax/Request.purs

```diff
@@ -0,0 +1,115 @@
+module Network.Affjax.Request
+  ( Ajax()
+  , AjaxRequest()
+  , MethodName(..)
+  , Content(..)
+  , AjaxResponse()
+  , defaultRequest
+  , ajax
+  ) where
+
+import Control.Monad.Aff (Aff(), EffA(), makeAff)
+import Control.Monad.Eff (Eff())
+import Control.Monad.Eff.Exception(Error())
+import Data.Array ()
+import Data.ArrayBuffer.Types (ArrayView())
+import Data.Function (Fn8(), runFn8)
+import Data.Maybe (Maybe(..), maybe)
+import Data.Nullable (Nullable(), toNullable)
+import DOM (Document())
+import DOM.File (Blob())
+import DOM.XHR (FormData())
+import Network.HTTP (Header(..))
+
+-- | The event type for AJAX requests.
+foreign import data Ajax :: !
+
+-- | The parameters for an AJAX request.
+type AjaxRequest a =
+  { url :: String
+  , method :: MethodName
+  , headers :: [Header]
+  , content :: Maybe (Content a)
+  , username :: Maybe String
+  , password :: Maybe String
+  }
+
+-- | A HTTP method name: `GET`, `POST`, etc.
+newtype MethodName = MethodName String
+
+-- | The types of data that can be set in an AJAX request.
+data Content a
+  = ArrayViewContent (ArrayView a)
+  | BlobContent Blob
+  | DocumentContent Document
+  | TextContent String
+  | FormDataContent FormData
+
+-- TODO: probably not this? Do we want to deal with other responses, include headers, etc?
+newtype AjaxResponse = AjaxResponse String
```

Comment:
Yeah, we'll somehow want to support `responseBody`, `responseText`, and `responseXML`, since I think those are the 3 baked into `XMLHttpRequestObject`, together with headers and status code. If this is just a record, it's easy enough for users to extract out the appropriate field of interest, so including "too much information" (e.g. headers) is probably OK.